feat: add next-code-point-index string package (#1117)
stdlib-js/stdlib: lib/node_modules/@stdlib/string/next-code-point/README.md

<!--

@license Apache-2.0

Copyright (c) 2023 The Stdlib Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

-->

# nextCodePoint

> > Return the next code point in a string after a specified position.

Reviewer: ```suggestion
> Return the next Unicode code point in a string after a specified position.
```

> > Return the next code point in a string after a specified position.

Reviewer: @steff456  What is returned is not the next code point, but the next code point position. So we should update the descriptions here and elsewhere to

```
Return the position of the next Unicode code point in a string after a specified position.
```

Also, given the confusion of this package, we should rename the package to

```
next-code-point-index
```

with alias

```
nextCodePointIndex
```

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- Package usage documentation. -->

<section class="usage">

## Usage

```javascript
var nextCodePoint = require( '@stdlib/string/next-code-point' );
```

#### nextCodePoint( string\[, fromIndex] )

> Returns the next code point in a string after a specified position.

Reviewer: ```suggestion
Returns the next Unicode code point in a string after a specified position.
```

```javascript
var out = nextCodePoint( 'last man standing' );
// returns 1
```

> By default, the function searches for a grapheme cluster starting from the first index. To specify an alternative starting search index, provide a `fromIndex` argument.

Reviewer: ```suggestion
By default, the function searches for a Unicode code point starting from the first index. To specify an alternative starting search index, provide a `fromIndex` argument.
```
